src/winkerberos.c

```diff
@@ -419,6 +433,74 @@ sspi_client_init(PyObject* self, PyObject* args, PyObject* kw) {
     return resultobj;
 }
 
+static PyObject*
+sspi_server_init(PyObject* self, PyObject* args, PyObject* kw) {
+	sspi_server_state* state;
+	PyObject* pyctx = NULL;
+	PyObject* serviceobj;
+	PyObject* mechoidobj = Py_None;
+	WCHAR *service = NULL, *principal = NULL;
+	Py_ssize_t slen = 0;
+	WCHAR *mechoid = GSS_MECH_OID_KRB5;
+	PyObject* resultobj = NULL;
+	INT result = 0;
+	static SEC_CHAR* keywords[] = {
+		"service", "mech_oid", NULL };
+
+	if (!PyArg_ParseTupleAndKeywords(args,
+		kw,
+		"O|O",
+		keywords,
+		&serviceobj,
+		&mechoidobj)) {
```

Comment:
I believe Negotiate means the client can fallback to NTLM if Kerberos is unavailable. However using only Kerberos for SPNEGO resulted in ```AcquireCredentialsHandleW``` returning ```SEC_E_INSUFFICIENT_MEMORY```.

Reply:
> I believe Negotiate means the client can fallback to NTLM

That is what it means, but WinKerberos doesn't support NTLM.

> However using only Kerberos for SPNEGO resulted in AcquireCredentialsHandleW returning SEC_E_INSUFFICIENT_MEMORY

You mean when the client uses "Negotiate" WinKerberos errors in AcquireCredentialsHandle?

Reply:
> You mean when the client uses "Negotiate" WinKerberos errors in AcquireCredentialsHandle?

Yes, when the client used SPNEGO/Negotiate WinKerberos errors out in AcquireCredentialsHandle. There seem to be some fundamental difference between the two modes, even though Negotiate uses Kerberos in my case (I made sure I wasn't doing NTLM using wireshark).

Reply:
Then it sounds like we should hardcode L"Negotiate" into the AcquireCredentialsHandle call on the server side instead of supporting mechoid? If passing winkerberos.GSS_MECH_OID_KRB5 causes AcquireCredentialsHandle to always fail it doesn't seem very useful. :-)

Reply:
Perhaps using GSS_MECH_OID_KRB5 both on the client and server would make AcquireCredentialsHandle happy. Nothing I have tried but I believe AcquireCredentialsHandle would work correctly in that case.  I'm unsure if such an implementation is useful though.

Reply:
Yeah, probably not useful, since the server side won't know in advance. It's weird that you get SEC_E_INSUFFICIENT_MEMORY. I can't imagine why that would happen or not based on choice of security package.